### ISSUE TITLE
setParams triggers Vue warning: 'inject() can only be used inside setup() or functional components' when used after render

### ISSUE THREAD

**Issue by Ragura** [posva/vue-router-mock#179]
### Steps to reproduce the behavior

When testing components with this package as its mocked router and using `setParams()` after mounting (or in my case, rendering via Testing Library instead of Vue Test Utils) causes Vue to print the following warning: `inject() can only be used inside setup() or functional components`

The mocked router is set up as follows:
```
import {
  createRouterMock,
  injectRouterMock,
} from 'vue-router-mock'

const router = createRouterMock()
```

In the `beforeEach()` of the test suite:
```
beforeEach(async () => {
    injectRouterMock(router)
})
```

This code causes the warning:
```
it('Navigates to page if token was valid and user is fetched', async () => {
    await render(MyComponent, {
      global: {
        plugins: [
          router,
          createTestingPinia(),
        ],
      },
    })
    await router.setParams({ token: '123' }) // It's called after the render()
    await nextTick()
    expect(router.push).toHaveBeenCalledWith(
      expect.objectContaining({ name: 'pageName' }),
    )
  })
```
Moving the `setParams()` before the render removes the warning.
In other words, this code does **not** prompt the warning:
```
it('Navigates to page if token was valid and user is fetched', async () => {
    await router.setParams({ token: '123' }) // It's called before the render()
    // Render code here again
    await nextTick()
    expect(router.push).toHaveBeenCalledWith(
      expect.objectContaining({ name: 'pageName' }),
    )
  })
```

### Expected behavior

I'm not sure why the warning shows up, but it's not always possible to set every router param early on in the test.

**Comment by posva** [comment]
What is `render`? Usually, it's `mount` that needs to be called to properly setup the context. Note that `setParams()` (or any code here) calls `inject()`. You must use vue test utils too

**Comment by Ragura** [comment]
> What is `render`? Usually, it's `mount` that needs to be called to properly setup the context. Note that `setParams()` (or any code here) calls `inject()`. You must use vue test utils too

`render()` is [Testing Library's](https://github.com/testing-library/vue-testing-library) implementation of `mount()`. It takes the same global config (inherits it from VTU) and (should) executes it in the same manner. I can try converting it to Vue Test Utils' `mount()` function to see if it makes a difference. But if I understand correctly, it won't make a difference because `inject()` is called no matter what and so will print the warning?

**Comment by Ragura** [comment]
I have been able to pinpoint what's going on. It actually has nothing to do with the component directly. The component calls an action of a Pinia store. This Pinia store also imports and uses `useRouter()`. The store doesn't run inside the context of script setup, which means inject is not available there. Is there a way around this?

**Comment by posva** [comment]
try adding the router to stores like shown in pinia documentation. It’s normal what you describe doesn’t work
